spec/helpers/action-controller_spec.cr

```diff
@@ -0,0 +1,200 @@
+require "json"
+require "file_utils"
+require "action-controller"
+require "../spec_helper"
+require "../../src/openapi-generator/helpers/action-controller"
+
+class Payload
+  include JSON::Serializable
+  extend OpenAPI::Generator::Serializable
+
+  def initialize(@hello : String = "world")
+  end
+end
+
+class HelloPayloadActionController < ActionController::Base
+  include ::OpenAPI::Generator::Controller
+  include ::OpenAPI::Generator::Helpers::ActionController
+
+  base "/hello"
+
+  @[OpenAPI(
+    <<-YAML
+      summary: Sends a hello payload
+      responses:
+        200:
+          description: Overriden
+    YAML
+  )]
+  def create
+    query_params "mandatory", "A mandatory query parameter"
+    query_params? "optional", "An optional query parameter"
+
+    body_as Payload?, description: "A Hello payload."
+
+    payload = Payload.new
+    respond_with 200, description: "Hello" do
+      json payload, type: Payload
+      xml "<hello></hello>", type: String
+    end
+    respond_with 201, description: "Not Overriden" do
+      text "Good morning.", type: String
+    end
+    respond_with 400 do
+      text "Ouch.", schema: String.to_openapi_schema
+    end
+  end
+end
+
+require "../../src/openapi-generator/providers/action-controller.cr"
+
+OpenAPI::Generator::Helpers::ActionController.bootstrap
+
+describe OpenAPI::Generator::Helpers::ActionController do
+  after_all {
+    FileUtils.rm "openapi_test.yaml"
+  }
+
+  it "should infer the status codes and contents of the response body" do
+    options = {
+      output: Path[Dir.current] / "openapi_test.yaml",
+    }
+    base_document = {
+      info:       {title: "Test", version: "0.0.1"},
+      components: NamedTuple.new,
+    }
+
+    OpenAPI::Generator.generate(
+      OpenAPI::Generator::RoutesProvider::ActionController.new,
+      options: options,
+      base_document: base_document
+    )
+
+    openapi_file_contents = File.read "openapi_test.yaml"
+    openapi_file_contents.should eq <<-YAML
```

Comment:
Currently the `OpenAPI::Generator::Helpers::ActionController.bootstrap` method does create a YAML file, with the above example, the created YAML file is as follows 

```yaml
---
openapi: 3.0.1
info:
  title: Test
  version: 0.0.1
paths:
  /hello:
    get:
      summary: Sends a hello payload
      parameters:
      - name: mandatory
        in: query
        description: A mandatory query parameter
        required: true
        schema:
          type: string
      - name: optional
        in: query
        description: An optional query parameter
        required: false
        schema:
          type: string
      requestBody:
        description: The request payload.
        content:
          application/json:
            schema:
              allOf:
              - $ref: '#/components/schemas/Payload'
        required: false
      responses:
        "200":
          description: A hello payload?
          content:
            application/json:
              schema:
                allOf:
                - $ref: '#/components/schemas/Payload'
            application/xml:
              schema:
                type: string
        "201":
          description: A good morning message.
          content:
            text/plain:
              schema:
                type: string
        "400":
          description: Bad Request
          content:
            text/plain:
              schema:
                type: string
components:
  schemas:
    Payload:
      required:
      - hello
      type: object
      properties:
        hello:
          type: string
  responses: {}
  parameters: {}
  examples: {}
  requestBodies: {}
  headers: {}
  securitySchemes: {}
  links: {}
  callbacks: {}
```

However the following will not be added

```yaml
       parameters:
       - name: mandatory
         in: query
         description: A mandatory query parameter
         required: true
         schema:
           type: string
       - name: optional
         in: query
         description: An optional query parameter
         required: false
         schema:
           type: string
       requestBody:
         description: The request payload.
         content:
           application/json:
             schema:
               allOf:
               - $ref: '#/components/schemas/Payload'
         required: false
       responses:
         "201":
           description: A good morning message.
           content:
             text/plain:
               schema:
                 type: string
         "400":
           description: Bad Request
           content:
             text/plain:
               schema:
                 type: string
```